File: Kaskus/GridViewPaging/GridViewPagingVb/Pages/DataTableResult.vb
Imports System.Collections
Imports System.ComponentModel
Imports System.Data

Namespace Pages

    Public NotInheritable Class DataTableResult
        Inherits DataSourceResult
        Public Sub New()
            Records = New DataTable()
        End Sub

        Public Property Records() As DataTable

        Protected Overrides Function MyList() As IList
            Return (CType(Records, IListSource)).GetList()
        End Function
    End Class
End Namespace

File: Kaskus/GridViewPaging/GridViewPagingCs/Pages/DataTableResult.cs
namespace Dede.App.GridViewPaging.Pages
{
    using System.Collections;
    using System.ComponentModel;
    using System.Data;

    public sealed class DataTableResult
        : DataSourceResult
    {
        public DataTableResult()
        {
            this.Records = new DataTable();
        }

        public DataTable Records { get; set; }

        protected override IList MyList()
        {
            return ((IListSource)this.Records).GetList();
        }
    }
}